packages/files/directory.pony

```diff
@@ -1,12 +1,12 @@
 use "time"
 
-use @ponyint_o_rdonly[I32]()
-use @ponyint_o_rdwr[I32]()
-use @ponyint_o_creat[I32]()
-use @ponyint_o_trunc[I32]()
-use @ponyint_o_directory[I32]()
-use @ponyint_o_cloexec[I32]()
-use @ponyint_at_removedir[I32]()
+use @pony_o_rdonly[I32]()
+use @pony_o_rdwr[I32]()
+use @pony_o_creat[I32]()
+use @pony_o_trunc[I32]()
+use @pony_o_directory[I32]()
+use @pony_o_cloexec[I32]()
+use @pony_at_removedir[I32]()
```

Comment:
I agree with the three descriptions of the categories, but I don't necessarily think that we need to introduce a third naming conventions.  Here's how I think we should look at it:

- The name indicates the stability guarantee or lack thereof.
- The `PONY_API` macro or lack thereof dictates whether the symbol will be exported.

So, the three categories would then be:

- `PONY_API` + `pony_*`: public/stable API for users, exported to be accessible from JIT'd Pony programs because all Pony programs can reasonably expect this API to be available to them.
- `PONY_API` + `ponyint_*`: private/internal/unstable API - no packages outside of the standard library should be touching these, but they are exported so that they can be accessed by the standard library in JIT'd Pony programs.
- `ponyint_*`: private/internal/unstable API - no packages outside of the standard library should be touching these.  Non-JIT'd Pony programs will be able to "reach"/collide with these symbols (which is exactly the reason we added the prefix in the first place), but the same programs if compiled with the JIT will not.  But nobody should be using these from Pony programs anyway, because they are `ponyint_` and that's all you need to know to stay away from them.